source/beamline/CLS/CLSBeamline.cpp

```diff
@@ -1,12 +1,40 @@
 #include "CLSBeamline.h"
 
+#include "beamline/CLS/CLSStorageRing.h"
+
 CLSBeamline::CLSBeamline(const QString &controlName) :
 	AMBeamline(controlName)
 {
-
+	wasBeamAvailable_ = false;
 }
 
 CLSBeamline::~CLSBeamline()
 {
 
 }
+
+/**
+ * ================= implementation of public interface / slots =======================
+ **/
+
+void CLSBeamline::initializeBeamline()
+{
+	AMBeamline::initializeBeamline();
+
+	connect(CLSStorageRing::sr1(), SIGNAL(beamAvaliability(bool)), this, SLOT(updateBeamAvailabilityStatus()));
+
+	updateBeamAvailabilityStatus();
+}
+
+/**
+ * ================= implementation of protected slots / functions =======================
+ **/
+void CLSBeamline::updateBeamAvailabilityStatus()
```

Comment:
for BioXAS, PGM and SXRMB, it is called when the beamlineStatus_ value changed or connection changed; for VESPERS, it is called when the POE status changed.